src/config-array.js

```diff
@@ -759,6 +762,16 @@ export class ConfigArray extends Array {
 	 * @returns {Object} The config object for this file.
 	 */
 	getConfig(filePath) {
+		const config = this.getConfigOrNoMatch(filePath);
+		if (config !== NoMatchingConfig) return config;
+	}
+
+	/**
+	 * Returns the config object for a given file path or a `NoMatchingConfig` indicator.
+	 * @param {string} filePath The complete path of a file to get a config for.
+	 * @returns {Object} The config object or `NoMatchingConfig` indicator for this file.
+	 */
+	getConfigOrNoMatch(filePath) {
```

Comment:
This method doesn't need to be public, it could be converted into a local function.